### PR TITLE
[`pylint`] Implement `invalid-bytes-returned` (`E0308`)

### PR DESCRIPTION
**Pull request by tibor-reiss**
Add pylint rule invalid-bytes-returned (PLE0308)

See https://github.com/astral-sh/ruff/issues/970 for rules

Test Plan: `cargo test`

### CODE REVIEW

**Comment by tibor-reiss** [comment]
The tests run locally fine when using commit 670d66f as rebase, but there are failures when using the next commit, c221035. Was there some limit reached (rule is u16 in rule_set.rs / contains()) maybe?

File: crates/ruff_linter/src/rules/pylint/rules/invalid_bytes_return.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        return;
+    }
+
+    let returns = {
+        let mut visitor = ReturnStatementVisitor::default();
+        visitor.visit_body(body);
+        visitor.returns
+    };
+
+    for stmt in returns {
```
Comment (MichaReiser) [inline crates/ruff_linter/src/rules/pylint/rules/invalid_bytes_return.rs:60]: Should this rule and the other invalid-*-returned rules also flag the case where the method has no return statement

```
class Test:
	def __bytes__(self):
		print("nope")
```

**Review by MichaReiser — APPROVED** [review]

File: crates/ruff_linter/src/rules/pylint/rules/invalid_bytes_return.rs (hunk 1 above)
Comment (tibor-reiss) [reply crates/ruff_linter/src/rules/pylint/rules/invalid_bytes_return.rs:67]: I added this case as well. If this becomes the consensus, I can also add it to the already included bool (E0304) and str (E0307) cases.

Comment (tibor-reiss) [reply crates/ruff_linter/src/rules/pylint/rules/invalid_bytes_return.rs:75]: @MichaReiser As it turned out, the no-return case is not that simple: if there is just a pass/ellipsis/raise statement, then this rule should not raise. However, if there is e.g. a print statement and a raise statement, then it should raise - see the tests for examples.

**Review by charliermarsh — APPROVED** [review]

File: crates/ruff_linter/src/rules/pylint/rules/invalid_bytes_return.rs (hunk 1 above)
Comment (charliermarsh) [reply crates/ruff_linter/src/rules/pylint/rules/invalid_bytes_return.rs:73]: I changed this to use `is_stub` for detecting the other cases.

**Pull request merged**